### PR TITLE
course: add students as a multiline string

### PR DESCRIPTION
**Pull request by haraldschilly**
# Description

The goal of this is to make it easier to copy/paste student email addresses from a spreadsheet. Here are a couple of screenshots to explain what I mean.

libreoffice calc on my desktop:

screenshot from 2018-11-15 16-07-57

then paste this into the textarea form (it's possible to expand the area, by default it is 2 lines)

screenshot from 2018-11-15 16-10-49

that's how it looks with a slightly smaller box and the parsed/queried addresses in the confirmation list

screenshot from 2018-11-15 16-11-01

Regarding the code changes, the form is a bit different, because just changing the field type caused overlapping controls.

# Testing Steps
1. misc: `./src/smc-util$ export NODE_ENV=mocha-test && SMC_TEST=true node_modules/.bin/mocha --reporter ${REPORTER:-progress} test/misc-test.coffee`
   I've noticed that two unrelated tests fail for `misc.map_limit`. Which is weird, because the code is clearly not doing what the tests wants. I didn't find a documentation to know what's really right in order to fix this. I suspect the test is wrong, though.
1. repeat the steps above
1. make sure comma-separated list like `student1@cocla.com,student2@cocalc.com` also works. like so:
screenshot from 2018-11-15 16-17-55

**note**: this change in misc has postgres-query as a reverse depedency. hence making this work requires a hub restart.

# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):
- [ ] No debugging console.log messages.
- [ ] All new code is actually used.
- [ ] Non-obvious code has some sort of comments.

Front end:
- [ ] Restart at least one project and check its `~/.smc/local_hub/local_hub.log`
- [ ] Completely restart Webpack with `./w` in `/src`
- [x] Screenshots if relevant.

### CODE REVIEW

**Comment by williamstein** [comment]
> note: this change in misc has postgres-query as a reverse depedency. hence making this work requires a hub restart.

What happens if it is merged wihout a hub restart?  Does that cause bugs until the hub restart is done?

**Comment by haraldschilly** [comment]
well, the only change is splitting at linebreaks. without a hub restart it only splits at certain characters like a comma.

this is how it works without a hub restart (using the running one from working on jupyter)

base case, like before


screenshot from 2018-11-15 16-17-55

this also works, because it searches for all email addresses

screenshot from 2018-11-15 20-09-46

here it misses my name, though

screenshot from 2018-11-15 20-10-38

---

with the hub update, the last query looks for my name

screenshot from 2018-11-15 20-11-18

**Comment by williamstein** [comment]
So the answer to the question: "does it introduce bugs without a hub restart" is **YES**.  ?  

This is critical to know, since it impacts very much how I deploy this...

**Comment by haraldschilly** [comment]
no, no new bug, I mainly mentioned it to avoid forgetting to update the hub for a while.

**Comment by williamstein** [comment]
> no, no new bug,

hmmm.  But with the new multiline input, it seems like a user would expect it to work with multiline input...?

File: src/smc-webapp/course/students_panel.tsx [hunk 1]
```diff
@@ -200,7 +201,7 @@ export const StudentsPanel = rclass<StudentsPanelReactProps>(
       const { add_search } = this.state;
       return webapp_client.user_search({
         query: add_search,
-        limit: 50,
+        limit: 100,
```
[the remark this answers is not in the document]
Comment (haraldschilly) [reply src/smc-webapp/course/students_panel.tsx]: I was wondering about such a situation when I saw it. Maybe we should at least make a ticket to investigate what happens, and especially to see if a useful error message comes up all the time.

**Pull request merged**